File: lab 5/bubblesort.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
void bubbleSort(int A[], int n)
{
    int i, j, temp;
    for(i=0;i<n-1;i++)
        for(j=0;j<n-i-1;j++)
            if(A[j]>A[j+1])
            {
                temp = A[j];
                A[j] = A[j+1];
                A[j+1] = temp;
            }
}
void display(int A[], int n)
{
    int i;
    for(i=0;i<n;i++)
        printf("%d ",A[i]);
    printf("\n");
}
int main()
{
    int i,n;
    time_t start, end;
    int A[200000];
    printf("Enter n ");
    scanf("%d",&n);
    for(i=0;i<n;i++)
        A[i] = rand();
    display(A,n);
    start = time(NULL);
    bubbleSort(A,n);
    end = time(NULL);
    display(A,n);
    printf("The time taken is %.5f\n",difftime(end,start));
    return 0;
}

File: lab 5/bubblesort.py
import time
import random
def BubbleSort(A):
    n=len(A)
    i=0
    for i in range(n):
        for j in range(n-i-1):
           if(A[j]>A[j+1]):
               A[j],A[j+1]=A[j+1],A[j]
               j+=1
        i+=1 
n=int(input("Enter n "))
A=[random.randint(0,1000) for _ in range(n)]
print(A)
t1=time.time()
BubbleSort(A)
t2=time.time()
print(A)
print(f"Time taken is {t2-t1}")